kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/Type.java

```diff
@@ -24,4 +42,36 @@
 
     String toString( Object value );
 
+    public static Type determineTypeFromClass( Class<?> clazz ) {
+        if( clazz == null ) {
+            return BuiltInType.UNKNOWN;
+        } else if( Number.class.isAssignableFrom(clazz) ) {
+            return BuiltInType.NUMBER;
+        } else if( String.class.isAssignableFrom(clazz) ) {
+            return BuiltInType.STRING;
+        } else if( LocalDate.class.isAssignableFrom(clazz) ) {
+            return BuiltInType.DATE;
+        } else if( LocalTime.class.isAssignableFrom(clazz) || OffsetTime.class.isAssignableFrom(clazz) ) {
+            return BuiltInType.TIME;
+        } else if( ZonedDateTime.class.isAssignableFrom(clazz) || OffsetDateTime.class.isAssignableFrom(clazz) || LocalDateTime.class.isAssignableFrom(clazz) ) {
+            return BuiltInType.DATE_TIME;
+        } else if( Duration.class.isAssignableFrom(clazz) || Period.class.isAssignableFrom(clazz) ) {
+            return BuiltInType.DURATION;
+        } else if( Boolean.class.isAssignableFrom(clazz) ) {
+            return BuiltInType.BOOLEAN;
+        } else if( UnaryTest.class.isAssignableFrom(clazz) ) {
+            return BuiltInType.UNARY_TEST;
+        } else if( Range.class.isAssignableFrom(clazz) ) {
+            return BuiltInType.RANGE;
+        } else if( FEELFunction.class.isAssignableFrom(clazz) ) {
+            return BuiltInType.FUNCTION;
+        } else if( List.class.isAssignableFrom(clazz) ) {
+            return BuiltInType.LIST;
+        } else if( Map.class.isAssignableFrom(clazz) ) {     // TODO not so sure about this one..
```

Comment:
I remember we talked about this, but I'm wondering if at least at compiling phase it could do without having to rely con this `__TYPE__` convention. Technically, the compiler is given type hints in the form of {variable name, type}, so maybe this could be leveraged to avoid the use of mentioned convention.
The downside of the approach just described is that requires (manually) to specify nested Maps which should be treated as a typed-Map, and with their names (keys) regardless of the level; this is not the best either, I'm just thinking if there is a way to simplify also this last limitation.